mlflow/keras.py

```diff
@@ -385,33 +385,43 @@ class __MLflowKerasCallback(keras.callbacks.Callback):
         Records available logs after each epoch.
         Records model structural information as params when training begins
         """
+
         def on_train_begin(self, logs=None):  # pylint: disable=unused-argument
-            try_mlflow_log(mlflow.log_param, 'num_layers', len(self.model.layers))
-            try_mlflow_log(mlflow.log_param, 'optimizer_name', type(self.model.optimizer).__name__)
+            try_mlflow_log(
+                mlflow.log_param, 'num_layers', len(
+                    self.model.layers))
+            try_mlflow_log(
+                mlflow.log_param, 'optimizer_name', type(
+                    self.model.optimizer).__name__)
             if hasattr(self.model.optimizer, 'lr'):
                 lr = self.model.optimizer.lr if \
-                    type(self.model.optimizer.lr) is float \
+                    isinstance(self.model.optimizer.lr, float) \
                     else keras.backend.eval(self.model.optimizer.lr)
                 try_mlflow_log(mlflow.log_param, 'learning_rate', lr)
             if hasattr(self.model.optimizer, 'epsilon'):
                 epsilon = self.model.optimizer.epsilon if \
-                    type(self.model.optimizer.epsilon) is float \
+                    isinstance(self.model.optimizer.epsilon, float) \
                     else keras.backend.eval(self.model.optimizer.epsilon)
                 try_mlflow_log(mlflow.log_param, 'epsilon', epsilon)
 
             sum_list = []
             self.model.summary(print_fn=sum_list.append)
             summary = '\n'.join(sum_list)
-            try_mlflow_log(mlflow.set_tag, 'model_summary', summary)
-
-            tempdir = tempfile.mkdtemp()
-            try:
-                summary_file = os.path.join(tempdir, "model_summary.txt")
-                with open(summary_file, 'w') as f:
-                    f.write(summary)
-                try_mlflow_log(mlflow.log_artifact, local_path=summary_file)
-            finally:
-                shutil.rmtree(tempdir)
+            summary_file = "model_summary.txt"
+            if len(summary) <= 5000:
+                try_mlflow_log(mlflow.set_tag, 'model_summary', summary)
+            else:
+                try_mlflow_log(mlflow.set_tag, 'model_summary',
+                               'Model summary too long to be set as a tag. '
+                               'Please see the artifacts for the model summary.')
+                tempdir = tempfile.mkdtemp()
+                try:
+                    summary_file = os.path.join(tempdir, "model_summary.txt")
+                    with open(summary_file, 'w') as f:
+                        f.write(summary)
+                    try_mlflow_log(mlflow.log_artifact, local_path=summary_file)
+                finally:
+                    shutil.rmtree(tempdir)
```

Comment:
Sorry if I was unclear before, but we should be logging the model summary as an artifact regardless of whether or not the tag is longer than 5000 characters. You can move this out of the else block. 

Just to be clear, the only thing that should be changing here is what is being set as a tag. My previous comment about not writing "random files to the user's local directory" was referring to cleaning up the `tempdir` we create to order to write the artifact. Let me know if this makes sense!